format/Message.fbs

```diff
@@ -88,7 +252,8 @@ table DictionaryBatch {
 /// which may include experimental metadata types. For maximum compatibility,
 /// it is best to send data using RecordBatch
 union MessageHeader {
-  Schema, DictionaryBatch, RecordBatch, Tensor, SparseTensor
+  Schema, DictionaryBatch, RecordBatch, Tensor, SparseTensor, Digest,
```

Comment:
So a Digest is a separate message in the IPC stream? That sounds like a lot of overhead, no? Why not add it in the message itself?

Reply:
Two reasons:  
- If it is a seldom used feature, then I believe we pay an overhead for maintaining the null field on all messages.
- It becomes more complicated to describe how to calculate the digest because it becomes self referential.

Reply:
A null field is probably extremely cheap nowadays. The second reason (self-referential digest computation) is more convincing though.

Reply:
A problem here is that the receiver must wait for the next message before deciding whether to discard the current message or not. But the receiver doesn't know in advance if the sender is producing Digest messages. So the wait time for the next message is potentially unbounded. Why is that supposed to be solved?

Reply:
It seemed more natural to me having these after the messages.  I see two ways of doing it.
1.  Adding a field on Schema or using metadata to signal digests will be produced.
2.  Have digest messages always come before the message they validate.

Reply:
I changed this to propose digests are always for the next message in the stream.

Reply:
Also this begs the question: how are the file format (as opposed to the stream format) readers supposed to implement this feature? When they read a message, should they also seek to the previous message in the expectation that it may contain a digest for the message they're interested in?

Reply:
That is what I was thinking (I thought this would be a cheap operation, but we can add something to the footer otherwise).

Reply:
As discussed on call, I will try to add this directly to the message.

Reply:
Thinking about this a little bit more, I think keeping it as a separate message might make sense.  By embedding each message we need some convention for eliminating the hash from the message (i.e. zeroing or nulling the field).  The issue with this approach I think is it assumes the messages are mutable which won't always be the case (e.g. if we mmap a file we don't necessarily want to memcpy for mutable and we wouldn't want to touch the file).